### PR TITLE
add a transform in CCNode

### PR DESCRIPTION
**Pull request by jxhgzs**
...ship but we still need to maintain transformation  parent-child
        Example:
          GameObject_Parent(CCNode; create from CCNode)
                 GameObject_sub1(CCNode; Create from the same CCSpriteBatchNode,the node's parent is CCLayer )
                        RenderComponents1( Create form the same CCSpriteBatchNode)
                        Components2
                        ....
                 GameObject_sub2 (CCNode; Create from the CCNode,the node's parent is GameObject_Parent )
                       RenderComponents1( not used batchNode)
         Now, GameObject_sub1 not GameObject_Parent node, but I want it to simulate the change of the son is GameObject_Parent;
         I Need additional the father's matrix  in child Node ;

Example:(in cocos2d-x HelloWorldScene.cpp)
        //create batchNode
    CCSpriteBatchNode\* batch=  CCSpriteBatchNode::create("Icon.png");
    this->addChild(batch);
    //create sprite
    CCSprite\* parent = CCSprite::createWithTexture(batch->getTexture());
    CCSprite\* sub = CCSprite::create("Icon-72.png");  
    batch->addChild(parent);
    this->addChild(sub);  // it is not parent's sub

```
//position
parent->setPosition(ccp(200, 200));
CCAffineTransform t = parent->nodeToParentTransform();
sub->translateFromOtherNode(t);

//scale
parent->setScale(2);
t = parent->nodeToParentTransform();
sub->translateFromOtherNode(t);

//rotation
parent->setRotation(20);
t = parent->nodeToParentTransform();
sub->translateFromOtherNode(t);
```

### CODE REVIEW

File: cocos2dx/base_nodes/CCNode.cpp [hunk 1]
```diff
@@ -1158,13 +1159,17 @@ CCAffineTransform CCNode::nodeToParentTransform(void)
...
         }
-
+		m_sTransform = CCAffineTransformConcat(m_sTransform, m_sOtherTransform);
         m_bTransformDirty = false;
     }
 
     return m_sTransform;
 }
-
+void CCNode::translateFormOtherNode(CCAffineTransform &transform)
```
Comment (dumganhar) [inline cocos2dx/base_nodes/CCNode.cpp:1168]: translateFormOtherNode -> translateFromOtherNode?

Comment (jxhgzs) [inline cocos2dx/base_nodes/CCNode.cpp:1168]: translateFromOtherNode is right

Comment (dumganhar) [inline cocos2dx/base_nodes/CCNode.cpp:1168]: So you need to update this Pull Request.

Comment (dumganhar) [inline cocos2dx/base_nodes/CCNode.cpp:1168]: Do you think we renamed `translateFromOtherNode` to `setAdditionalTransform` is better?

Comment (jxhgzs) [inline cocos2dx/base_nodes/CCNode.cpp:1168]: right，  it is better；

**Pull request merged**